flink-yarn/src/main/java/org/apache/flink/yarn/AbstractYarnClusterDescriptor.java

```diff
@@ -131,6 +134,8 @@
 	/** Lazily initialized list of files to ship */
 	protected List<File> shipFiles = new LinkedList<>();
 
+	private List<URI> archives = new LinkedList<>();
```

Comment:
it is because we cannot know the size in advance, and there are usually little elements for archives, so I just keep the same structure with shipFiles.